back-end/src/routes/api.js

```diff
@@ -104,8 +97,203 @@ const uploadFile = (request, response) => {
+            photo: photoData.Body.buffer,
+            title: items[i].title,
+            desc: items[i].desc,
+            likes: items[i].likes,
+            uploadDate: items[i].uploadDate
+          };
+
+          responseData.push(responseObject);
+          prevTitle = items[i].title;
+        }
+        statusCode = 200;
+        console.debug(responseData);
+        response.status(statusCode).send(responseData);
+      }
+      catch(e) {
+        console.debug("Could not retrieve photos", e);
+      }
+      console.debug("Retrieval successful");
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+  })().catch(e => {
+    console.debug("Something went wrong");
+  });
+};
+
+/**
+ * Retrieve all photos from a specific tag.
+ * @param {tag: String} request 
+ * @param {[{photo: Object, uploadDate: String, title: String}]} response 
+ */
+const getPhotoByTag = (request, response) => {
+  (async () => {
+    var statusCode = 400;
+    var message = "Empty";
+    const reqTag = request.headers.tag;
+    if (reqTag) {
+      try {
+        var queryParams = {
+          TableName: "Files",
+          IndexName: "tag-file-index",
+          KeyConditionExpression: "#tag = :tag",
+          ExpressionAttributeNames: {
+            "#tag": "tag"
+          },
+          ExpressionAttributeValues: {
+            ":tag": reqTag
+          }
+        };
+
+        const queryData = await dynamodb.query(queryParams).promise();
+        const items = queryData.Items;
+        console.debug("Data Queried");
+        console.debug(items);
 
+        var fileKeys = [];
+
+        var responseData = [];
+        // Loop through photos to send all object data
+        for (let i = 0; i < items.length; i++) {
+          var getParams = {
+            Bucket: BUCKET_NAME,
+            Key: items[i].file
+          };
+          var photoData = await s3.getObject(getParams).promise();
+            
+          responseObject = {
+            photo: photoData.Body.buffer,
+            title: items[i].title,
+            uploadDate: items[i].uploadDate
+          };
+
+          responseData.push(responseObject);
+        }
+        statusCode = 200;
+        console.debug(responseData);
+        response.status(statusCode).send(responseData);
+      }
+      catch(e) {
+        console.debug("Could not retrieve information", e);
+      }
+    }
+    else {
+      response.status(statusCode).send(message);
+    }
+  })().catch(e => {
+    console.debug("Something went wrong");
+  });
+};
+
+/**
+ * Retrieve all tags and one of their associated photos. 
+ * @param {userID: Integer} request 
+ * @param {[{tag: String, photo: Object}]} response 
+ */
+const getTags = (request, response) => {
+  (async () => {
+    var statusCode = 400;
+    var message = "Nothing to see here folks";
+    // I assume this line is supposed to check if the user is authenticated. Replace once authentication is done.
+    if (true) {
```

Comment:
why?

Reply:
That can probably be removed. On the document it said the request takes in a userID. I'm assuming that means the user must be authenticated for this to be called. I just left that in as a placeholder before authentication is implemented.

Reply:
remove please

Reply:
remove please